### PR TITLE
Pre workshop polish

### PR DESCRIPTION
**Pull request by StrawWagen**
General polish of copying materials.
Can now copy ent's "innate" materials.
+Allow admins to change advanced material of players.

Fixed the preview not drawing sometimes.
Fixed a potential crash issue with the preview, according to the wiki at least.

### CODE REVIEW

File: lua/autorun/sh_mateditor.lua [hunk 1]
```diff
@@ -63,6 +63,7 @@ end
 -- indexes are 1 / 3 to catch advmat 2'ed props, removed index 2 support, it was too specialized.
 local alphaTypes = {
 	[1] = "$alphatest",
+	[3] = "$vertexalpha",
```
Comment (wrefgtzweve) [inline lua/autorun/sh_mateditor.lua]: index 3 already exists in this table

File: lua/weapons/gmod_tool/stools/advmat.lua [hunk 2]
```diff
@@ -40,9 +40,22 @@ TOOL.Information = {
 	MATERIALIZE
 */
 
+local canAdvmatPlayers = CreateConVar( "advmat_canmaterializeplayers", "1", { FCVAR_ARCHIVE, FCVAR_REPLICATED }, "Can admins change the Advanced Material of players?", 0, 1 )
+
+-- admins can advmat players
+function TOOL:LegalMaterialize( trace )
```
Comment (wrefgtzweve) [inline lua/weapons/gmod_tool/stools/advmat.lua:46]: this should not be up to the tool, this should respect CPPI

File: lua/weapons/gmod_tool/stools/advmat.lua [hunk 3]
```diff
@@ -247,15 +308,23 @@ if CLIENT then
...
+
+		local ent = eyeTr.Entity
 
 		if not IsValid( ent ) then return end
-		local mat = tobool( toolObj:GetClientInfo( "usenoise" ) ) and toolObj.Preview or toolObj.PreviewMat
+		local mat = toolObj:GetPreviewMat()
+
+		-- according to DrawModel on wiki this will fix a crash
+		local entsFlags = ent:GetEFlags()
+		if bit.band( entsFlags, EF_BONEMERGE ) ~= 0 then return end
```
Comment (wrefgtzweve) [inline lua/weapons/gmod_tool/stools/advmat.lua]: you can use IsEffectActive here

File: lua/autorun/sh_mateditor.lua (hunk 1 above)
Comment (StrawWagen) [reply lua/autorun/sh_mateditor.lua]: Fixed

File: lua/weapons/gmod_tool/stools/advmat.lua (hunk 3 above)
Comment (StrawWagen) [reply lua/weapons/gmod_tool/stools/advmat.lua]: Suggestion is applied.

**Pull request merged**